packages/x-components/src/x-modules/url/components/url-handler.vue

```diff
@@ -1,82 +1,209 @@
      *
-     * @returns A boolean whether the parameter is present or not.
+     * @param url - The URL to remove parameters from.
+     * @internal
+     * **/
+    protected deleteUrlParameters(url: URL): void {
+      this.paramsNames.forEach(paramName => url.searchParams.delete(this.getUrlKey(paramName)));
+    }
+
+    /**
+     * Set all the provided parameters to the url with the mapped key.
+     *
+     * @param url - The current URL.
+     * @param urlParams - The list of parameters to add.
+     * @internal
+     * **/
+    protected setUrlParameters(url: URL, urlParams: UrlParams): void {
+      const sortedParameters = this.sortParams(urlParams);
+      sortedParameters.forEach(([paramName, paramValue]) => {
+        const urlParamKey = this.getUrlKey(paramName);
+        if (Array.isArray(paramValue)) {
+          paramValue.forEach(value => {
+            url.searchParams.append(urlParamKey, String(value));
+          });
+        } else {
+          url.searchParams.set(urlParamKey, String(paramValue));
+        }
+      });
+    }
+
+    /**
+     * Sorts the params in a tuple array [key,value] to generate always the same URL with the params
+     * in the same order.
+     *
+     * @param urlParams - The {@link UrlParams} to sort.
+     * @returns An array of tuples with the key-value of each paramter, sorted by key.
      * @internal
      */
-    protected get isQueryInUrl(): boolean {
-      const urlParams = new URLSearchParams(document.location.search);
-      return urlParams.has(this.$attrs.query ?? 'query');
+    protected sortParams(urlParams: UrlParams): Array<[string, unknown]> {
+      return Object.entries(urlParams).sort(([param1], [param2]) => {
+        return param1 < param2 ? -1 : 1;
+      });
     }
 
     /**
-     * Open X if there is a query in the Url.
+     * Returns the URL param value parsed depending on its type in the initial store state. As we
+     * can not know what type can have an extra param, all extra params are parsed as strings. We
+     * know if it is an extra param because it is not in the initial state.
      *
+     * @param urlSearchParams - The {@link URLSearchParams} to get the param.
+     * @param paramName - The name of the param in {@link UrlParams}.
+     * @returns The parsed value.
      * @internal
      */
-    protected emitOpenEvent(): void {
-      if (this.isQueryInUrl) {
-        this.$x.emit('UserOpenXProgrammatically');
+    protected getParamByType(
+      urlSearchParams: URLSearchParams,
+      paramName: string
+    ): UrlParamValue | undefined {
+      const key = this.getUrlKey(paramName);
+      if (!(paramName in initialUrlState.params)) {
+        return urlSearchParams.get(key) ?? undefined;
+      } else {
+        switch (typeof initialUrlState.params[paramName]) {
+          case 'number': {
+            const numberOrNull = urlSearchParams.get(key);
+            return numberOrNull ? Number(numberOrNull) : undefined;
```

Comment:
we have a rule to avoid the use of it:
![image](https://user-images.githubusercontent.com/5759712/137314398-b32db90b-7633-4f2c-b2ba-56672e627d9e.png)